### PR TITLE
geometric means for lineplot yaxis

### PR DESCRIPTION
**Pull request by d-callan**

### CODE REVIEW

**Comment by d-callan** [comment]
based on comments from @SheenaTomko ive put something together here that will err if it sees negative values as input to a request for geometric means and will ignore 0 values if present. modifying that logic is pretty simple so im happy to get this out and revise as necessary later once we have broader feedback.

File: R/bin.R [hunk 1]
```diff
@@ -68,6 +68,41 @@ binMean <- function(data, x, y, group = NULL, panel = NULL, binWidth = NULL, vie
...
+  } else {
+    data <- data[order(data$binLabel),]
+  }
+  data$binLabel <- as.character(data$binLabel)
+
+  data <- collapseByGroup(data, group, panel)
+
+  return(data)
+}
+
```
Comment (asizemore) [inline R/bin.R:105]: For later: eventually may be worth collapsing all these bin<Strategy> functions to one

File: tests/testthat/test-utils.R [hunk 2]
```diff
@@ -8,6 +8,24 @@ test_that("chiSq returns consistent classes", {
   expect_equal(all(unlist(lapply(chiSq(tbl2),is.numeric))), TRUE)
 })
 
+test_that("roundedGeometricMean() returns reasonable results.", {
+  noValues <- numeric()
+  singlePositiveValue <- rnorm(1, 10)
+  fewPositiveValues <- rnorm(2, 10)
```
Comment (asizemore) [inline tests/testthat/test-utils.R:14]: fewPositiveValues not used in the tests just below?

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -56,6 +74,51 @@ test_that("*CI fxns return reasonable results", {
...
   expect_equal(medianCI(manyValues)[[1]]$upperBound < max(manyValues), TRUE)
 
+  singlePositiveValue <- rnorm(1, 10)
+  fewPositiveValues <- rnorm(2, 10)
+  negativeValues <- rnorm(1000, -10)
+  positiveValues <- rnorm(1000, 10)
+  zeroValue <- c(0, positiveValues)
+  onlyZeroValues <- c(0,0,0,0,0)
+
+  # should this return a graceful nonspecific error like the other types of inputs we cant find CIs for?
```
Comment (asizemore) [inline tests/testthat/test-utils.R:84]: It just says "no those are negative values, no?". Seems good to me

File: tests/testthat/test-utils.R [hunk 4]
```diff
@@ -56,6 +74,51 @@ test_that("*CI fxns return reasonable results", {
...
+
+  expect_equal(nchar(geometricMeanCI(zeroValue)[[1]]$error) == 0, TRUE)
+  expect_equal(is.numeric(geometricMeanCI(zeroValue)[[1]]$lowerBound), TRUE)
+  expect_equal(is.numeric(geometricMeanCI(zeroValue)[[1]]$upperBound), TRUE)
+  expect_equal(geometricMeanCI(zeroValue)[[1]]$lowerBound <= roundedGeometricMean(zeroValue), TRUE)
+  expect_equal(geometricMeanCI(zeroValue)[[1]]$upperBound >= roundedGeometricMean(zeroValue), TRUE)
+  expect_equal(geometricMeanCI(zeroValue)[[1]]$lowerBound > min(zeroValue), TRUE)
+  expect_equal(geometricMeanCI(zeroValue)[[1]]$upperBound < max(zeroValue), TRUE)
+  expect_warning(geometricMeanCI(zeroValue))
+
```
Comment (asizemore) [inline tests/testthat/test-utils.R:121]: may also be worth checking if `geometricMeanCI(zeroValue) == geometricMeanCI(positiveValues)`

**Review by asizemore — APPROVED** [review]
looks good to me!

**Pull request merged**